examples/flutter_gallery/lib/demo/pesto_demo.dart

```diff
@@ -176,6 +177,83 @@ class _RecipeGridPageState extends State<RecipeGridPage> {
   }
 }
 
+class PestoLogo extends StatefulWidget {
+  PestoLogo({this.showText, this.height});
+
+  final bool showText;
+  final double height;
+
+  @override
+  _PestoLogoState createState() => new _PestoLogoState();
+}
+
+class _PestoLogoState extends State<PestoLogo> {
+  // Native sizes for logo and its image/text components.
+  static const double kLogoHeight = 162.0;
+  static const double kLogoWidth = 220.0;
+  static const double kImageHeight = 108.0;
+  static const double kTextHeight = 48.0;
+  final TextStyle titleStyle = const PestoStyle(fontSize: kTextHeight, fontWeight: FontWeight.w900, color: Colors.white, letterSpacing: 3.0);
+
+  AnimationController _textController;
+  Animation<double> _textOpacity;
+  Animation<double> _imageHeight;
+
+  @override
+  void initState() {
+    super.initState();
+    _textController = new AnimationController(duration: const Duration(milliseconds: 200))
+      ..value = config.showText ? 1.0 : 0.0;
+    _textOpacity = new CurvedAnimation(parent: _textController.view, curve: const Interval(0.3, 1.0, curve: Curves.easeInOut));
+    _imageHeight = new Tween<double>(begin: kLogoHeight, end: kImageHeight).animate(
+      new CurvedAnimation(parent: _textController.view, curve: const Interval(0.0, 0.9, curve: Curves.easeInOut))
+    );
+    _imageHeight.addListener(() {
+      setState(() {
+        // Logo height has changed.
+      });
+    });
+  }
+
+  @override
+  void didUpdateConfig(PestoLogo oldConfig) {
+    if (config.showText != oldConfig.showText) {
+      if (config.showText)
+        _textController.forward();
+      else
+        _textController.reverse();
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Transform(
+      transform: new Matrix4.identity()
+                   ..scale(config.height / kLogoHeight),
+      alignment: FractionalOffset.topCenter,
+      child: new SizedBox(
+        width: kLogoWidth,
+        child: new Stack(
+          overflow: Overflow.visible,
+          children: <Widget>[
+            new Positioned(
+              left: 0.0, top: 0.0, height: _imageHeight.value, width: kLogoWidth,
```

Comment:
If you could use `new Positioned.fromRect(rect: _imageRectAnimation.value)` here